cmd/crank/beta/trace/trace.go

```diff
@@ -38,19 +39,21 @@ import (
 )
 
 const (
-	errGetResource        = "cannot get requested resource"
-	errCliOutput          = "cannot print output"
-	errKubeConfig         = "failed to get kubeconfig"
-	errInitKubeClient     = "cannot init kubeclient"
-	errGetDiscoveryClient = "cannot get discovery client"
-	errGetMapping         = "cannot get mapping for resource"
-	errInitPrinter        = "cannot init new printer"
+	errGetResource            = "cannot get requested resource"
+	errCliOutput              = "cannot print output"
+	errKubeConfig             = "failed to get kubeconfig"
+	errInitKubeClient         = "cannot init kubeclient"
+	errGetDiscoveryClient     = "cannot get discovery client"
+	errGetMapping             = "cannot get mapping for resource"
+	errInitPrinter            = "cannot init new printer"
+	errMissingName            = "missing name, must be provided separately 'TYPE[.VERSION][.GROUP] [NAME]' or in the 'TYPE[.VERSION][.GROUP][/NAME]' format"
+	errInvalidResourceAndName = "invalid resource and name"
 )
 
 // Cmd builds the trace tree for a Crossplane resource.
 type Cmd struct {
-	Resource string `arg:"" help:"Kind of the of the Crossplane resource, accepts the 'TYPE[.VERSION][.GROUP]' format."`
-	Name     string `arg:"" help:"Name of the Crossplane resource."`
+	Resource string `arg:"" help:"Kind of the of the Crossplane resource, accepts the 'TYPE[.VERSION][.GROUP][/NAME]' format."`
+	Name     string `arg:"" optional:"" help:"Name of the Crossplane resource."`
```

Comment:
As I was saying I didn't want to put much emphasis on the naming, it's optional and the regex shows that you can put it in the first argument too, I feel it could be enough. Otherwise I would have to add a comment saying ", can be also passed as part of the first argument." Or some longer version of it... I think the docs can show that it works both ways and we can keep it short here.

Reply:
gave it a try, LMK: https://github.com/crossplane/crossplane/pull/4910/commits/e0622f4d4d213291f4d28775dbc3c4c985b236d6